platforms/sdl/main.cpp

```diff
@@ -15,6 +15,8 @@ typedef AppPlatform_sdl UsedAppPlatform;
 #include "client/app/NinecraftApp.hpp"
 #include "client/player/input/Multitouch.hpp"
 
+#undef main // anti-SDL2
```

Comment:
SDL2 defines main and replaces it with SDL_main. As far as I know, this is unnecessary and breaks SDL2 linking on Windows.

Reply:
Unless you're actually building with /subsystem:console I don't see why it would cause linking errors, but if it works, it works.

Reply:
> SDL2 defines main and replaces it with SDL_main.

Yeah, it does. And that's on purpose. Are you sure you're linking to SDL correctly? (Including `SDL2main.lib`?)

See relevant StackOverflow: https://stackoverflow.com/a/11976179/16198887.

Reply:
That's exactly what I checked. I guess I didn't consider the stuff sdl_main was doing relevant.

Reply:
The way I see it, while `#undef main` might technically work now in this situation, it is the incorrect way of starting an SDL application. And it may break in the future and/or on other (non-Windows) platforms. IMO it is better to do things the correct way and get `SDL_main` to work.